lib/minidoc/finders.rb

```diff
@@ -33,19 +33,21 @@ def find(id_or_selector, options = {})
       end
     end
 
-    def find_one(selector, options = {})
+    def find_one(selector = {}, options = {})
       wrap(collection.find_one(selector, options))
     end
 
-    def find_one!(query, options = {})
-      find_one(query, options) or raise DocumentNotFoundError
+    def find_one!(selector = {}, options = {})
+      find_one(selector, options) or raise DocumentNotFoundError
     end
 
-    def find_one_or_initialize(attributes, options = {})
-      raise ArgumentError unless attributes.is_a?(Hash)
-      find_one(attributes, options) || new(attributes)
+    def find_one_or_initialize(selector = {}, options = {})
```

Comment:
Nit: I think `attributes` was actually the right name here. We're also using it as a selector, but it's constrained by what's valid for attributes (which is a subset of what's valid for a selector).

e.g. `{ foo: "bar" }` is fine as an arg to this method, but `{ age: { "$gt": 20 } }` is nonsensical.